ext/openssl/extconf.rb

```diff
@@ -13,7 +13,16 @@
 
 require "mkmf"
 
-dir_config_given = dir_config("openssl").any?
+openssl_dirs = dir_config("openssl")
+dir_config_given = openssl_dirs.any?
+
+_, ldir = openssl_dirs
+if ldir&.split(File::PATH_SEPARATOR)&.none? { |dir| File.directory?(dir) }
+  raise "OpenSSL library directory could not be found in '#{ldir}'. " \
+    'You might want to use --with-openssl-lib=<dir> option to specify the ' \
+    'directory.'
+end
```

Comment:
I referred to the this error message from other part in the `ext/openssl/extconf.rb` below.

```
if !pkg_config_found && !find_openssl_library
  Logging::message "=== Checking for required stuff failed. ===\n"
  Logging::message "Makefile wasn't created. Fix the errors above.\n"
  raise "OpenSSL library could not be found. You might want to use " \
    "--with-openssl-dir=<dir> option to specify the prefix where OpenSSL " \
    "is installed."
end
```